crates/topos/src/components/node/mod.rs

```diff
@@ -0,0 +1,132 @@
+use clap::{CommandFactory, Parser};
+use opentelemetry::global;
+use tokio::{
+    signal, spawn,
+    sync::{mpsc, oneshot},
+};
+
+use figment::error::Kind;
+use std::path::Path;
+use std::{
+    fs::{create_dir_all, File, OpenOptions},
+    io::Write,
+    str::FromStr,
+};
+
+use tracing::{error, info};
+use tracing_opentelemetry::OpenTelemetrySpanExt;
+
+use self::commands::{NodeCommand, NodeCommands};
+
+use crate::config::sequencer::SequencerConfig;
+use crate::config::tce::TceConfig;
+
+use crate::{
+    config::{base::BaseConfig, insert_into_toml, load_config, node::NodeConfig, Config},
+    tracing::setup_tracing,
+};
+
+pub(crate) mod commands;
+
+pub(crate) async fn handle_command(
+    NodeCommand {
+        subcommands,
+        verbose: _,
+        home,
+    }: NodeCommand,
+) -> Result<(), Box<dyn std::error::Error>> {
+    match subcommands {
+        Some(NodeCommands::Init(cmd)) => {
+            let name = cmd.name.expect("No name or default was given");
+
+            // Construct path to node config
+            // will be $TOPOS_HOME/node/default/ with no given name
+            // and $TOPOS_HOME/node/<name>/ with a given name
+            let node_path = home.join("node").join(name);
```

Comment:
At first I think we don't need to generate the `$TOPOS_HOME/node/default/config.toml` but the `$TOPOS_HOME/node/base.toml` right @hadjiszs ?

Reply:
The read of a specific node already works. But I don't build the `base.toml`, just the specific `<node_name>/config.toml`. I mark this therefore as resolved.